DICOM/src/Microsoft.Health.DeID.SharedLib/RedactFunction.cs

```diff
@@ -0,0 +1,166 @@
+                _ => null,
+            };
+        }
+
+        public uint? Redact(uint value, AnonymizerValueTypes valueType)
+        {
+            return valueType switch
+            {
+                AnonymizerValueTypes.Age => RedactAge(value),
+                _ => null,
+            };
+        }
+
+        public decimal? Redact(decimal value, AnonymizerValueTypes valueType)
+        {
+            return valueType switch
+            {
+                AnonymizerValueTypes.Age => RedactAge(value),
+                _ => null,
+            };
+        }
+
+        public DateTimeOffset? Radact(DateTimeOffset dateTime)
+        {
+            EnsureArg.IsNotNull<DateTimeOffset>(dateTime, nameof(dateTime));
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                if (DateTimeUtility.IndicateAgeOverThreshold(dateTime))
+                {
+                    return null;
+                }
+
+                return new DateTimeOffset(dateTime.Year, 1, 1, 0, 0, 0, default);
+            }
+
+            return null;
+        }
+
+        public DateTimeObject Redact(DateTimeObject dateObject)
+        {
+            EnsureArg.IsNotNull(dateObject, nameof(dateObject));
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                if (DateTimeUtility.IndicateAgeOverThreshold(dateObject.DateValue))
+                {
+                    return null;
+                }
+
+                dateObject.DateValue = new DateTimeOffset(dateObject.DateValue.Year, 1, 1, 0, 0, 0, dateObject.HasTimeZone == true ? dateObject.DateValue.Offset : new TimeSpan(0, 0, 0));
+                return dateObject;
+            }
+
+            return null;
+        }
+
+        public AgeObject Redact(AgeObject age)
+        {
+            EnsureArg.IsNotNull(age, nameof(age));
+
+            return RedactAge(age.AgeInYears()) == null ? null : age;
+        }
+
+        private string RedactDate(string dateString)
+        {
+            if (string.IsNullOrEmpty(dateString))
+            {
+                return null;
+            }
+
+            DateTimeOffset date = DateTimeUtility.ParseDateTimeString(dateString, DeIDGlobalSettings.DateFormat);
+
+            return Radact(date)?.ToString(DeIDGlobalSettings.DateFormat);
+        }
+
+        private string RedactDateTime(string dateTimeString)
+        {
+            if (string.IsNullOrEmpty(dateTimeString))
+            {
+                return null;
+            }
+
+            DateTimeOffset date = DateTimeUtility.ParseDateTimeString(dateTimeString, DeIDGlobalSettings.DateTimeFormat);
+
+            return Radact(date)?.ToString(DeIDGlobalSettings.DateTimeFormat);
+        } 
+
+        private uint? RedactAge(uint age)
+        {
+            return (uint?)RedactAge((decimal)age);
+        }
+
+        private decimal? RedactAge(decimal age)
+        {
+            if (_redactSetting.EnablePartialAgesForRedact)
```

Comment:
🟢 [Suggestion]

A single if sentence is good enough, like Qiufeng mentioned in this file.